docs/developer/development/ci.rst

```diff
@@ -0,0 +1,147 @@
+Continuous Testing
+==================
+
+Add a new test in the continuous integration system
+---------------------------------------------------
+
+A test as continuous integration system level is an End-to-End test that need a
+dedicated environment (with one or several machines (virtual or container)).
+
+A test that only check a specific feature of a classic MetalK8s deployment
+should be part of PyTest BDD and not integrated as a dedicated stage in
+continuous integration system.
+
+How to choose between Pre-merge and Post-merge
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Choice really depend on the goal of this test.
+
+As high level view:
+
+Pre-merge:
+
+- Test is not too long (less than 30 minutes).
+- Test essential feature of the product (installation, expansion, building,
+  ...).
+
+Post-merge:
+
+- Test last longer (more than 30 minutes).
+- Test "non-essential" feature of the product (upgrade, downgrade, solutions,
+  ...).
+
+How to add a stage in continuous integration system
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Continuous integration system is controled by the ``eve/main.yml`` YAML file.
+
+A stage is defined by a worker and a list of steps. Each stage should be in
+``stages`` section and trigger by ``pre-merge`` or ``post-merge``.
```

Comment:
Each stage should be in the stages section and is triggered by pre-merge or post-merge